### PR TITLE
buildah 0.8

### PR DESCRIPTION
**Pull request by jimmyjones2**
- Don't use interpolation (move to env)
- Easier way of specifying build args - previously would have to use `BUILD_EXTRA_ARGS` and quote them very carefully
- Add warning for `BUILD_EXTRA_ARGS` and `PUSH_EXTRA_ARGS` that inputs need to be sanitized 

e.g. now the following are correctly passed as Dockerfile build arguments:
```
        - name: BUILD_ARGS
          value:
            - |
                foo=b\"a'z"u uu
            - baz=8888$(echo foo)ssss`ssss` -v a:b
```

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by jimmyjones2** [comment]
@vdemeester @vinamra28 Can you review please?

**Review by vdemeester — APPROVED** [review]
cc @vinamra28

**Comment by jimmyjones2** [comment]
@vinamra28 Converted to v1

File: task/buildah/0.8/buildah.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    description: Extra parameters passed for the push command when pushing images. WARNING - must be sanitized to avoid command injection
+    type: string
+    default: ""
+  - name: SKIP_PUSH
+    description: Skip pushing the built image
+    default: "false"
+  - name: BUILD_ARGS
+    description: Dockerfile build arguments, array of key=value
+    type: array
+    default:
```
Comment (vinamra28) [inline task/buildah/0.8/buildah.yaml:59]: `BUILD_EXTRA_ARGS` is already there, do we still need this?

Comment (jimmyjones2) [reply task/buildah/0.8/buildah.yaml:59]: Yes, this is so people can pass arbitrary extra command line options to the buildah build command

Comment (vinamra28) [reply task/buildah/0.8/buildah.yaml:59]: ack,lastly, can you please squash the commits? will merge it

**Comment by jimmyjones2** [comment]
@vinamra28 Squashed and ready to merge :)

**Pull request merged**